link/link_other.go

```diff
@@ -167,6 +201,118 @@ func (kpm *KprobeMultiInfo) Missed() (uint64, bool) {
 	return kpm.missed, kpm.count > 0
 }
 
+type KprobeMultiAddress struct {
+	Address uint64
+	Cookie  uint64
+}
+
+// Addresses are the addresses hooked by the kprobe.
+func (kpm *KprobeMultiInfo) Addresses() ([]KprobeMultiAddress, bool) {
+	if kpm.addrs == nil || len(kpm.addrs) != len(kpm.cookies) {
+		return nil, false
+	}
+	addrs := make([]KprobeMultiAddress, len(kpm.addrs))
+	for i := range kpm.addrs {
+		addrs[i] = KprobeMultiAddress{
+			Address: kpm.addrs[i],
+			Cookie:  kpm.cookies[i],
+		}
+	}
+	return addrs, true
+}
+
+// Cookies are the cookies for each address hooked by the kprobe.
+func (kpm *KprobeMultiInfo) Cookies() ([]uint64, bool) {
+	return kpm.cookies, kpm.cookies != nil
+}
+
+type UprobeMultiInfo struct {
+	count         uint32
+	flags         uint32
+	missed        uint64
+	offsets       []uint64
+	cookies       []uint64
+	refCtrOffsets []uint64
+	path          string
+	pid           uint32
+}
+
+// AddressCount is the number of addresses hooked by the kprobe.
+func (umi *UprobeMultiInfo) AddressCount() (uint32, bool) {
+	return umi.count, umi.count > 0
+}
+
+func (umi *UprobeMultiInfo) Flags() (uint32, bool) {
+	return umi.flags, umi.flags > 0
+}
+
+func (umi *UprobeMultiInfo) Missed() (uint64, bool) {
+	return umi.missed, umi.missed > 0
+}
+
+type UprobeMultiOffset struct {
+	Offset uint64
+	Cookie uint64
+	RefCtr uint64
+}
+
+// Offsets returns the offsets that the uprobe was attached to along with the related cookies and ref counters.
+func (umi *UprobeMultiInfo) Offsets() ([]UprobeMultiOffset, bool) {
+	if umi.offsets == nil || len(umi.cookies) != len(umi.offsets) || len(umi.refCtrOffsets) != len(umi.offsets) {
+		return nil, false
+	}
+	var adresses = make([]UprobeMultiOffset, len(umi.offsets))
+	for i := range umi.offsets {
+		adresses[i] = UprobeMultiOffset{
+			Offset: umi.offsets[i],
+			Cookie: umi.cookies[i],
+			RefCtr: umi.refCtrOffsets[i],
+		}
+	}
+	return adresses, true
+}
+
+// Symbols returns the symbols that the uprobe was attached to.
+func (umi *UprobeMultiInfo) Symbols() ([]UprobeSymbol, error) {
+	if umi.offsets == nil {
+		return nil, fmt.Errorf("no offsets available")
+	}
+	ex, err := OpenExecutable(umi.path)
```

Comment:
hmm I wonder if this is safe to do; should we just return addresses like for kprobeMulti?